packages/auth/src/core/auth/password_policy_impl.ts

```diff
@@ -0,0 +1,192 @@
+      this.customStrengthOptions.containsUppercaseLetter =
+        responseOptions.containsUppercaseCharacter;
+    }
+    if (responseOptions.containsNumericCharacter !== undefined) {
+      this.customStrengthOptions.containsNumericCharacter =
+        responseOptions.containsNumericCharacter;
+    }
+    if (responseOptions.containsNonAlphanumericCharacter !== undefined) {
+      this.customStrengthOptions.containsNonAlphanumericCharacter =
+        responseOptions.containsNonAlphanumericCharacter;
+    }
+
+    this.enforcementState = response.enforcementState;
+    if (this.enforcementState === 'ENFORCEMENT_STATE_UNSPECIFIED') {
+      this.enforcementState = 'OFF';
+    }
+
+    // Use an empty string if no non-alphanumeric characters are specified in the response.
+    this.allowedNonAlphanumericCharacters =
+      response.allowedNonAlphanumericCharacters?.join('') ?? '';
+
+    this.forceUpgradeOnSignin = response.forceUpgradeOnSignin ?? false;
+    this.schemaVersion = response.schemaVersion;
+  }
+
+  validatePassword(password: string): PasswordValidationStatus {
+    const status: PasswordValidationStatusInternal = {
+      isValid: true,
+      passwordPolicy: this
+    };
+
+    // Check the password length and character options.
+    this.validatePasswordLengthOptions(password, status);
+    this.validatePasswordCharacterOptions(password, status);
+
+    // Combine the status into single isValid property.
+    status.isValid &&= status.meetsMinPasswordLength ?? true;
+    status.isValid &&= status.meetsMaxPasswordLength ?? true;
+    status.isValid &&= status.containsLowercaseLetter ?? true;
+    status.isValid &&= status.containsUppercaseLetter ?? true;
+    status.isValid &&= status.containsNumericCharacter ?? true;
+    status.isValid &&= status.containsNonAlphanumericCharacter ?? true;
+
+    return status;
+  }
+
+  /**
+   * Validates that the password meets the length options for the policy.
+   *
+   * @param password Password to validate.
+   * @param status Validation status.
+   */
+  private validatePasswordLengthOptions(
+    password: string,
+    status: PasswordValidationStatusInternal
+  ): void {
+    const minPasswordLength = this.customStrengthOptions.minPasswordLength;
+    const maxPasswordLength = this.customStrengthOptions.maxPasswordLength;
+    if (minPasswordLength) {
+      status.meetsMinPasswordLength = password.length >= minPasswordLength;
+    }
+    if (maxPasswordLength) {
+      status.meetsMaxPasswordLength = password.length <= maxPasswordLength;
+    }
+  }
+
+  /**
+   * Validates that the password meets the character options for the policy.
+   *
+   * @param password Password to validate.
+   * @param status Validation status.
+   */
+  private validatePasswordCharacterOptions(
+    password: string,
+    status: PasswordValidationStatusInternal
+  ): void {
+    // Assign statuses for requirements even if the password is an empty string.
+    this.updatePasswordCharacterOptionsStasuses(
```

Comment:
Statuses is spelled incorrectly for this method.  It should be `updatePasswordCharacterOptionsStatuses`

BTW, I had to do a triple check because even the correct spelling of `Statuses` doesn't look right to me, either. :)

Reply:
Whoops, not sure how I missed that. I fixed it, thanks.